packages/smooth_app/lib/pages/smooth_bottom_navigation_bar.dart

```diff
@@ -1,85 +1,109 @@
 import 'package:flutter/material.dart';
-import 'package:smooth_app/pages/history_page.dart';
 import 'package:smooth_app/pages/scan/scan_page.dart';
-import 'package:smooth_app/pages/user_preferences_page.dart';
-
-class _Page {
-  const _Page({required this.name, required this.icon, required this.body});
-  final String name;
-  final IconData icon;
-  final Widget body;
-}
+import 'package:smooth_app/widgets/tab_navigator.dart';
 
 enum SmoothBottomNavigationTab {
   Profile,
   Scan,
   History,
 }
 
-class SmoothBottomNavigationBar extends StatelessWidget {
-  const SmoothBottomNavigationBar({
-    this.tab = _defaultTab,
-  });
-
-  final SmoothBottomNavigationTab tab;
-
-  static const SmoothBottomNavigationTab _defaultTab =
-      SmoothBottomNavigationTab.Scan;
+class SmoothBottomNavigationBar extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => SmoothBottomNavigationBarState();
+}
 
-  static const List<SmoothBottomNavigationTab> _tabs =
-      <SmoothBottomNavigationTab>[
+class SmoothBottomNavigationBarState extends State<SmoothBottomNavigationBar> {
+  List<SmoothBottomNavigationTab> pageKeys = <SmoothBottomNavigationTab>[
     SmoothBottomNavigationTab.Profile,
     SmoothBottomNavigationTab.Scan,
     SmoothBottomNavigationTab.History,
   ];
+  SmoothBottomNavigationTab _currentPage = SmoothBottomNavigationTab.Scan;
 
-  static const Map<SmoothBottomNavigationTab, _Page> _pages =
-      <SmoothBottomNavigationTab, _Page>{
-    SmoothBottomNavigationTab.Profile: _Page(
-      name: 'Profile', // TODO(monsieurtanuki): translate
-      icon: Icons.account_circle,
-      body: UserPreferencesPage(),
-    ),
-    SmoothBottomNavigationTab.Scan: _Page(
-      name: 'Scan or Search',
-      icon: Icons.search,
-      body: ScanPage(),
-    ),
-    SmoothBottomNavigationTab.History: _Page(
-      name: 'History',
-      icon: Icons.history,
-      body: HistoryPage(),
-    ),
+  final Map<SmoothBottomNavigationTab, GlobalKey<NavigatorState>>
+      _navigatorKeys = <SmoothBottomNavigationTab, GlobalKey<NavigatorState>>{
+    SmoothBottomNavigationTab.Profile: GlobalKey<NavigatorState>(),
+    SmoothBottomNavigationTab.Scan: GlobalKey<NavigatorState>(),
+    SmoothBottomNavigationTab.History: GlobalKey<NavigatorState>(),
   };
+  int _selectedIndex = 1;
```

Comment:
What's the difference between `_currentPage` and `_selectedIndex`?

Reply:
There isn't a big difference, we only need it since the BottomNavigationBar needs a `currentIndex` int.
Of course we could run everything via the index, but I decided to use the `_currentPage` to keep it easier to read.

Reply:
That's what I suspected. It's not easier to read `_selectTab(_pageKeys[index], index)` when you ask yourself: there are 2 parameters, that means something, perhaps it's possible to write `_selectTab(_pageKeys[0], 1)` in some strange cases.
I guess either `BottomNavigationTab get _currentPage => _pageKeys[_selectedIndex]` or the other way around would be easier to read.

Reply:
Updated it, now we just use the index of _currentPage instead of a own var